packages/market/src/blockchain-facade/MarketUser.ts

```diff
@@ -37,12 +39,25 @@ export class Entity extends User.Entity {
 
         return new Entity(this.id, this.configuration).sync();
     }
+
+    async sync(): Promise<Entity> {
+        return super.sync();
```

Comment:
Previously `MarketUser.sync()` returned `User.Entity` instead of `MarketUser.Entity`. This is a fix that improves types and correctly returns `MarketUser.Entity` from `MarketUser.Entity.sync()` function.